### PR TITLE
Use system robot accounts

### PR DESCRIPTION
**Pull request by j-zimnowoda**
Context: 
https://app.zenhub.com/workspaces/dev-5e7e84dddc966f05a627a19b/issues/redkubes/unassigned-issues/353

End result:
image

### CODE REVIEW

File: src/tasks/harbor/harbor.ts [hunk 1]
```diff
@@ -84,39 +89,46 @@ const systemRobot: any = {
 
 const robotPrefix = 'otomi-'
 const config: any = {
+  // eslint-disable-next-line @typescript-eslint/camelcase
```
Comment (Morriz) [inline src/tasks/harbor/harbor.ts:92]: can we just put this to the top of the file? repeating the same exclusion so many times is unnecessary

File: src/tasks/harbor/harbor.ts [hunk 2]
```diff
@@ -2,13 +2,18 @@ import {
   ConfigureApi,
   HttpBearerAuth,
   MemberApi,
+  // eslint-disable-next-line no-unused-vars
```
Comment (Morriz) [inline src/tasks/harbor/harbor.ts:5]: can we just put this to the top of the file? repeating the same exclusion so many times is unnecessary

Comment (Morriz) [reply src/tasks/harbor/harbor.ts:5]: and why not remove unused vars?

**Review by Morriz — COMMENTED** [review]
just the redundant lint exclusions should be cleaned up imo then its fine

File: src/tasks/harbor/harbor.ts (hunk 1 above)
Comment (j-zimnowoda) [reply src/tasks/harbor/harbor.ts:92]: sure, then we will ignore camelcase  in the whole file

File: src/tasks/harbor/harbor.ts (hunk 2 above)
Comment (j-zimnowoda) [reply src/tasks/harbor/harbor.ts:5]: I you search through the file you will find out that it is used.
Then linter may have a bug

Comment (0-sv) [reply src/tasks/harbor/harbor.ts:7]: I think @Morriz means you can add a global eslint flag in the first line of the files, then it gets applied to all lines.

File: .env.sample [hunk 3]
```diff
@@ -9,7 +9,7 @@ HARBOR_PASSWORD=''
 HARBOR_BASE_URL='http://127.0.0.1:8083'
 HARBOR_BASE_REPO_URL='harbor.dev.eks.otomi.cloud'
 OIDC_CLIENT_SECRET=''
-OIDC_ENDPOINT='https://keycloak.dev.gke.otomi.cloud'
+OIDC_ENDPOINT='https://keycloak.dev.gke.otomi.cloud/realms/otomi'
```
Comment (0-sv) [inline .env.sample:12]: I am confused, I think we already do a string replace in our gotmpl logic wrt `/realms/otomi` (e.g. line 8 harbor.gotmpl)?

File: src/tasks/harbor/harbor.ts [hunk 4]
```diff
@@ -190,58 +203,61 @@ async function ensureSystemSecret(): Promise<RobotSecret> {
...
+  bearerAuth.accessToken = robotSecret.secret
+  return bearerAuth
 }
 
-async function ensureProjectSecret(teamId: string, projectId: string): Promise<void> {
-  const namespace = `team-${teamId}`
-
-  let k8sSecret = (await getSecret(projectSecretName, namespace)) as RobotSecret
+/**
+ * Ensure that Harbor robot account and corresponding Kubernetes pull secret exist
```
Comment (0-sv) [inline src/tasks/harbor/harbor.ts:211]: I see you're adding function docs, that's awesome, but do we use something like JSdoc or do you do it manually?

File: src/tasks/harbor/harbor.ts (hunk 2 above)
Comment (j-zimnowoda) [reply src/tasks/harbor/harbor.ts:7]: ah, noe we do not want to disable this check for the whole file

File: .env.sample (hunk 3 above)
Comment (j-zimnowoda) [reply .env.sample:12]: this is env.sample that intended to be use if you run task without otomi-core

File: src/tasks/harbor/harbor.ts (hunk 4 above)
Comment (j-zimnowoda) [reply src/tasks/harbor/harbor.ts:211]: It is added automatically while you start typing `/**`

**Comment by Morriz** [comment]
I say we wait of my other PR and merge that first as it solves issues related to testing

**Review by 0-sv — APPROVED** [review]
lgtm

**Comment by j-zimnowoda** [comment]
@Morriz ?

**Pull request merged**